app/src/org/commcare/activities/CommCareActivity.java

```diff
@@ -406,6 +406,11 @@ public void taskCancelled() {
 
     public void cancelCurrentTask() {
         stateHolder.cancelTask();
+

```

Comment:
This actually needs to fire the cancel signal, _and_ validate whether the signal can be processed (IE: Are we past the threshold where we're already processing)